### ISSUE TITLE
ScssPHP appears to fail silently

### ISSUE THREAD

**Issue by HLSiira** [scssphp/scssphp#80]
Hello,
I'm trying to implement a server-side compiling of SCSS using scssphp, for use in [this plugin](https://github.com/HLSiira/Atheos-SassCompiler). I know that the plugin works, and I know that my server-side php works. There are no errors in my apache log, and I'm beginning to think that I missed something on the installation somehow. Here is the code I'm using so far, it's straight from the website:

```
require_once "scssphp/scss.inc.php";
use ScssPhp\ScssPhp\Compiler;

try {
$scss = new Compiler();

echo $scss->compile('$color: #abc; div { color: lighten($color, 20%); }');
} catch (\Exception $e) {
    echo '';
    syslog(LOG_ERR, 'scssphp: Unable to compile content');
}
```

I've tried the latest pull from the master repo, and tried the 1.0.7 and 1.0.6 releases and nothing is returned to the client. Other variations I've used and their results:

```
require_once "scssphp/scss.inc.php";
use ScssPhp\ScssPhp\Compiler;

$scss = new Compiler();
echo $scss->compile('$color: #abc; div { color: lighten($color, 20%); }');
// Returns nothing to the client
```

```
require_once "scssphp/scss.inc.php";
use ScssPhp\ScssPhp\Compiler;
echo json_encode(array("status" => "success", "content" => __DIR__));
// Returns the json_encode to the client, exactly as it should.
```

I've even pushed that json_encode echo down into the scss.inc.php and it is included correctly and echoes correctly. Whenever I try to actually compile the scss, the compiler returns nothing. I'm running it on a Debian9 server with php 7.0+ something running. Please let me know if I'm just being an idiot, because I feel like it's user error, but I'm out of ideas. Thank you.

**Comment by allejo** [comment]
Make sure PHP is reporting errors:

```php
ini_set('display_errors', 1);
ini_set('display_startup_errors', 1);
error_reporting(E_ALL);
```

**Comment by HLSiira** [comment]
Done, reports nothing. I even broke some other lines to double check and those errors report correctly. I'm going to go out on a limb and ask, is there anything that should have been installed along with or already with ScssPHP?

Code I'm using:
```
require_once "scssphp/scss.inc.php";
use ScssPhp\ScssPhp\Compiler;

if (isset($_GET['path'])) {
	$path = $_GET['path'];
	// echo json_encode(array("status" => "success", "path" => $path));
	ini_set('display_errors', 1);
	ini_set('display_startup_errors', 1);
	error_reporting(E_ALL);

	$scss = new Compiler();
	echo $scss->compile('
		$color: #abc;
		div { color: lighten($color, 20%); }
	');
} else {
	echo '{"status":"error","message":"Missing Parameter!"}';
}
```

Thank you for your help btw.

**Comment by robocoder** [comment]
I changed the `require_once` path and your example works for me.

Have you tried running `composer install` and then the unit tests to see what fails in your deployment environment?

**Comment by HLSiira** [comment]
No, I'm not using composer, and I'm trying not to use it. I'm trying to build a plugin for an IDE, so drag/drop the plugin folder in and it works; so having composer or anything like that wouldn't work. It really limits what I can do but it's how the system is set up.

**Comment by robocoder** [comment]
I suggested using composer in order to run scssphp's unit tests and thus, help troubleshoot the issue you're encountering.

**Comment by HLSiira** [comment]
I used composer, and set up the tests. I received an Orange "ok, but some tests were skipped" as a response and nothing else. I'm starting to think that i just need to rebuilt my server stack as maybe I'm missing some small part of PHP. I'm still running 7.0.something but from what I understand, that shouldn't be causing an issue with ScssPHP

**Comment by robocoder** [comment]
Perhaps there's a difference in your conf.d between your web server (e.g., apache) and cli configuration.

Closing this for now. Please open a new ticket against 1.0.8 if you're still running into this issue after updating your server stack.

**Comment by HLSiira** [comment]
In case someone else ever finds this issue and can't figure it out, I solved this issue by realizing that I'm an idiot. My goal with this project was to update an already existing plugin. I was so focused on the backend that I didn't notice the front end was using jquery's getJSON. I was returning plain CSS, and jquery likes to silently fail  because CSS doesn't parse into a JSON value. Why jquery siliently fails instead of actually informing the user of an error, like any good piece of software would, is beyond me.

Running ScssPhp would work perfectly every single time and I wasted several hours of my life. If you  are using jquery and find this issue, I hope I saved you some time.
